### DISCUSSION TITLE
Calculated state in RepeatableEntry

### DISCUSSION THREAD

**Discussion by djl997** [filamentphp/filament#14773] [Help]
### Package

Infolist builder

### Package Version

v3.2.123

### How can we help you?

In this example of filament docs, we can multiply record amount and vat_rate:
```php
Infolists\Components\TextEntry::make('amount_including_vat')
    ->state(function (Model $record): float {
        return $record->amount * (1 + $record->vat_rate);
})
```
Ref: https://filamentphp.com/docs/3.x/infolists/entries/getting-started#calculated-state

But in my project, the TextEntry is in a RepeatableEntry. I have multiple products of value and quantity, and I want to display the total price per product row. I cannot get this to work...

```php
Infolists\Components\RepeatableEntry::make('products')->schema([
    Infolists\Components\TextEntry::make('value')
        ->money('EUR'),
    Infolists\Components\TextEntry::make('quantity'),
    Infolists\Components\TextEntry::make('total') // value * quantity
        ->money('EUR')
        // 1
        ->state(function($record) {
            // $record returns record, but not the repeated row??
        })
        // 2
        ->state(function(Infolists\Components\TextEntry $textEntry) {
            // $textEntry returns this field, and via ->getContainer() I can access the row
            $textEntry->getContainer()->getComponents()[0] //value field
            $textEntry->getContainer()->getComponents()[1] //quantity field
            // ... but this doesn't feel right, because, what if I change the order of fields? And furthermore, this doesn't feel as readable as filament intended, I think...
        }),
])
```

I can't find any documentation on how to achieve this. I would also be helped if there is somewhere a list of parameters that I can use in the `state()` function.

**Comment by SerjoA** [comment]
your solution was super helpful so thank u

**Comment by djl997** [comment]
A small followup on this issue, in v4 we have a similar situation, see https://filamentphp.com/docs/4.x/forms/repeater#adding-extra-item-actions-to-a-repeater for the documented solution. 

```php
function (array $arguments, Repeater $component): void {
    $itemData = $component->getItemState($arguments['item']);

    // e.g. $itemData['email']
}
```

Fortunately, the items are now fetchable by name, but it still feels like a technical intermediate step.
